### PR TITLE
feat: migrated latest fixes

### PR DESCRIPTION
**Pull request by adubovik**
* feat: added timings to debug prints
* feat: allowed empty messages for Bison models
* feat: supported streaming for Bison models
* fix: disabled malfunctioning clustering- and classification- gecko endpoints
* chore: bumped version of aidial-sdk to 0.1.2
* feat: supported history truncation via max_prompt_tokens/discarded_messages parameters

### CODE REVIEW

File: client/chat.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        return cls(model)
+
+    async def send_message(
+        self, prompt: str, params: ModelParameters, usage: TokenUsage
+    ) -> AsyncGenerator[str, None]:
+        self.history.append(
+            VertexAIMessage(author=VertexAIAuthor.USER, content=prompt)
+        )
+
+        consumer: Optional[Consumer] = None
```
Comment (Oleksii-Klimov) [inline client/chat.py]: Optional[CollectConsumer]?

File: aidial_adapter_vertexai/llm/consumer.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from aidial_adapter_vertexai.universal_api.token_usage import TokenUsage
+
+
+class Consumer(ABC):
+    @abstractmethod
+    async def append_content(self, content: str):
+        pass
+
+    @abstractmethod
+    async def add_usage(self, usage: TokenUsage):
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/llm/consumer.py]: Maybe set_usage(), given that it's supposed to be called once?

File: aidial_adapter_vertexai/universal_api/token_usage.py [hunk 3]
```diff
@@ -19,19 +9,10 @@ class TokenUsage(BaseModel):
...
-        ret: TokenUsageDict = {
-            "prompt_tokens": self.prompt_tokens,
-            "total_tokens": self.total_tokens,
-        }
-
-        if self.completion_tokens != 0:
-            ret["completion_tokens"] = self.completion_tokens
-
-        return ret
+    def __add__(self, other: "TokenUsage") -> "TokenUsage":
```
Comment (Oleksii-Klimov) [inline aidial_adapter_vertexai/universal_api/token_usage.py]: Is this method really needed? It looks like the way it is used now can be replaced with the accumulate method.

**Review by Oleksii-Klimov — DISMISSED** [review]

File: client/chat.py (hunk 1 above)
Comment (adubovik) [reply client/chat.py]: Fixed

File: aidial_adapter_vertexai/llm/consumer.py (hunk 2 above)
Comment (adubovik) [reply aidial_adapter_vertexai/llm/consumer.py]: Fixed

File: aidial_adapter_vertexai/universal_api/token_usage.py (hunk 3 above)
Comment (adubovik) [reply aidial_adapter_vertexai/universal_api/token_usage.py]: Fixed. Removed.

**Review by Oleksii-Klimov — APPROVED** [review]

**Pull request merged**